XenAdmin/Wizards/NewPolicyWizard/NewPolicyWizardSpecific.cs

```diff
@@ -0,0 +1,303 @@
+                {
+                    xenTabPageFinish.Summary = GetSummary();
+                    xenTabPageFinish.SelectedVMsCount = xenTabPageVMsPage.SelectedVMs.Count;
+                }
+            }
+            else
+            {
+                if (prevPageType == typeof(NewPolicySnapshotFrequencyPage))
+                {
+                    xenTabPageFinish.Summary = GetSummary();
+                    xenTabPageFinish.SelectedVMsCount = xenTabPageVMsPage.SelectedVMs.Count;
+                }
+                else if (prevPageType == typeof(NewVMGroupVMsPage<T>))
+                {
+                    xenTabPageSnapshotType.SelectedVMs = xenTabPageVMsPage.SelectedVMs;
+                }
+            }
+
+        }
+
+        protected override void FinishWizard()
+        {
+            if (VMGroup<T>.isVMPolicyVMPP)
+            {
+                var vmpp = new VMPP
+                {
+                    name_label = xenTabPagePolicy.PolicyName,
+                    name_description = xenTabPagePolicy.PolicyDescription,
+                    backup_type = xenTabPageSnapshotType.BackupType,
+                    backup_frequency = xenTabPageSnapshotFrequency.Frequency,
+                    backup_schedule = xenTabPageSnapshotFrequency.Schedule,
+                    backup_retention_value = xenTabPageSnapshotFrequency.BackupRetention,
+                    archive_frequency = xenTabPageArchive.ArchiveFrequency,
+                    archive_target_config = xenTabPageArchive.ArchiveConfig,
+                    archive_target_type = xenTabPageArchive.ArchiveTargetType,
+                    archive_schedule = xenTabPageArchive.Schedule,
+                    is_alarm_enabled = xenTabPageEmail.EmailEnabled,
+                    alarm_config = xenTabPageEmail.EmailSettings,
+                    is_policy_enabled = xenTabPageVMsPage.SelectedVMs.Count == 0 ? false : true,
+                    Connection = Pool.Connection
+                };
+
+                var action = new CreateVMPP(vmpp, xenTabPageVMsPage.SelectedVMs, xenTabPageFinish.RunNow);
+                action.RunAsync();
+            }
+            else
+            {
+                var vmss = new VMSS
+                {
+                    name_label = xenTabPagePolicy.PolicyName,
+                    name_description = xenTabPagePolicy.PolicyDescription,
+                    type = xenTabPageSnapshotType.BackupTypeVMSS,
+                    frequency = (vmss_frequency)xenTabPageSnapshotFrequency.Frequency,
+                    schedule = xenTabPageSnapshotFrequency.Schedule,
+                    retained_snapshots = xenTabPageSnapshotFrequency.BackupRetention,
+                    enabled = xenTabPageVMsPage.SelectedVMs.Count == 0 ? false : true,
+                    Connection = Pool.Connection
+                };
+
+                var action = new CreateVMSS(vmss, xenTabPageVMsPage.SelectedVMs, xenTabPageFinish.RunNow);
```

Comment:
I think the CreateVMPP and CreateVMSS blocks can also be combined, via a VMGroup<T>.CreateObjectAction() function. We will pass in the archive data even for VMSS, and it will get ignored later.